app/jobs/pubsub_job.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class PubsubJob < ApplicationJob
+  # self.queue_adapter = :pubsub
+  retry_on(StandardError, wait: 5.second, attempts: 2)
+
+  def perform(*args)
+    ActiveSupport::Notifications.instrument("perform.pubsub_job") do
```

Comment:
You don't have to instrument - the instrumentation already exists, you can subscribe instead:

https://guides.rubyonrails.org/active_support_instrumentation.html#active-job

Reply:
Ah, you did subscribe, my oversight. For perform, there's already built-in instrumentation in ActiveJob, so you just need to subscribe.

Reply:
Got this. I added another commit https://github.com/skysemilla/kisi-api-challenge/pull/1/commits/9a21d53a0541cc08008bc1441cee7f3ef974f5bd to use the built-in intrumentation for ActiveJob methods.